### PR TITLE
Removed Ord/PartialOrd derive rules for types.

### PR DESCRIPTION
**Pull request by sanket1729**
- Added custom Eq/PartialEq for miniscript
- Removed Ord/PartialOrd derive rules for types.

### CODE REVIEW

File: src/miniscript/types/correctness.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@
 use super::{ErrorKind, Property};
 
 /// Basic type representing where the fragment can go
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
```
Comment (apoelstra) [inline src/miniscript/types/correctness.rs]: Why remove these?

File: src/miniscript/types/mod.rs [hunk 2]
```diff
@@ -31,10 +31,12 @@ fn return_none<T>(_: usize) -> Option<T> {
 }
 
 /// Detailed type of a typechecker error
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
 pub enum ErrorKind {
     /// Relative or absolute timelock had a time value of 0
     ZeroTime,
+    /// Passed a `z` arguement to a `d` wrapeer when `z` was expected
+    NonZeroDupIf,
```
Comment (apoelstra) [inline src/miniscript/types/mod.rs]: Heh, arguably this should be in a separate commit, but I'll allow it

Comment (sanket1729) [reply src/miniscript/types/mod.rs]: I added it to the other merged PR. Forgot to remove it from here

Comment (sanket1729) [reply src/miniscript/types/mod.rs]: Don't know why this is not creating a conflict

Comment (apoelstra) [reply src/miniscript/types/mod.rs]: If they're exactly the same change git is smart enough to unify them.

**Pull request merged**